pyqtgraph/graphicsItems/TargetItem.py

```diff
@@ -1,132 +1,429 @@
+            self.currentBrush = self.hoverBrush
+            self.currentPen = self.hoverPen
+        else:
+            self.currentBrush = self.brush
+            self.currentPen = self.pen
+        self.update()
 
     def hoverEvent(self, ev):
-        if self.movable:
-            ev.acceptDrags(QtCore.Qt.LeftButton)
+        if self.movable and (not ev.isExit()) and ev.acceptDrags(QtCore.Qt.LeftButton):
+            self.setMouseHover(True)
+        else:
+            self.setMouseHover(False)
+
+    def viewTransformChanged(self):
+        GraphicsObject.viewTransformChanged(self)
+        self._shape = None  # invalidate shape, recompute later if requested.
+        self.update()
+
+    def pos(self):
+        """Provides the current position of the TargetItem
+
+        Returns
+        -------
+        tuple
+            ``(x, y)`` position of the TargetItem
+        """
+        return self._pos
+
+    def label(self):
+        """Provides the TargetLabel if it exists
+
+        Returns
+        -------
+        TargetLabel or None
+            If a TargetLabel exists for this TargetItem, return that, otherwise
+            return None
+        """
+        return self._label
+
+    def setLabel(self, text=None, labelOpts=None):
+        """Method to call to enable or disable the TargetLabel for displaying text
+
+        Parameters
+        ----------
+        text : Callable or str, optional
+            Details how to format the text, by default None
+            If None, do not show any text next to the TargetItem
+            If Callable, then the label will display the result of ``text(x, y)``
+            If a fromatted string, then the output of ``text.format(x, y)`` will be
+            displayed
+            If a non-formatted string, then the text label will display ``text``, by
+            default None
+        labelOpts : dictionary, optional
+            These arguments are passed on to :class:`~pyqtgraph.TextItem`
+        """
+        if text is None and self._label is not None:
+            # remove the label if it's already added
+            if self._label.scene() is not None:
+                self._label.scene().removeItem(self._label)
+            self._label = None
+        else:
+            # convert None entry for text to False
+            text = False if text is None else text
+
+            # provide default text if text is True
+            if isinstance(text, bool):
+                # convert to default value or empty string
+                text = "x = {: .3n}\ny = {: .3n}" if text else ""
+
+            # if there is no label, add a label
+            if self._label is None:
+                labelOpts = {} if labelOpts is None else labelOpts
+                self._label = TargetLabel(self, text=text, **labelOpts)
+
+            # if there is already a label, update the text formatting
+            else:
+                # update the label text
+                self._label.setFormat(text)
```

Comment:
We maybe also need to do something with the `labelOpts`.

Reply:
Updating the label options isn't on its own already supported by the labels. We could destroy the current label and make a new one, as a cheap implementation

Reply:
re-creation a TargetLabel here would actually be clever; as trying to support the mapping of all the individual options supported would be pretty tricky, and I'll likely screw up something.

If a user wanted to preserve the original item, they could likely just call `tragetItem.label().<whatevermethod>`

I think that's what I'll do...